crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1044,6 +1059,8 @@ impl<'db> TypeInferenceBuilder<'db> {
             }
         }
 
+        let decorator_tys = decorator_tys.into_boxed_slice();
```

Comment:
nit: there's only one usage of `decorator_tys` after this point, shadowing it seems unnecessary; why not just change `decorator_tys` to `decorator_tys.into_boxed_slice()` below?